### ISSUE TITLE
no_prepare_ocamlcocci.ml compile error on rocky linux 8

### ISSUE THREAD

**Issue by sahlberg** [coccinelle/coccinelle#345]
Trying to build the current master branch on rocky linux 8 (~centos 8)
but it fails with a type error:

OCAMLOPT  ocaml/[no_prepare_ocamlcocci.ml](http://no_prepare_ocamlcocci.ml/)
File "ocaml/[no_prepare_ocamlcocci.ml](http://no_prepare_ocamlcocci.ml/)", line 140, characters 29-36:
140 |      mcodefn mcodefn mcodefn mcodefn mcodefn mcodefn
                                   ^^^^^^^
Error: The function applied to this argument has type
         ?meta_mcode:(Ast_cocci.meta_name, 'a) V.cmcode ->
         ?string_mcode:(string, 'a) V.cmcode ->
         ?const_mcode:(Ast_cocci.constant, 'a) V.cmcode ->
         ?simpleAssign_mcode:(Ast_cocci.simpleAssignOp, 'a) V.cmcode ->
         ?opAssign_mcode:(Ast_cocci.arithOp, 'a) V.cmcode ->
         ?fixOp_mcode:(Ast_cocci.fixOp, 'a) V.cmcode ->
         ?unaryOp_mcode:(Ast_cocci.unaryOp, 'a) V.cmcode ->
         ?arithOp_mcode:(Ast_cocci.arithOp, 'a) V.cmcode ->
         ?logicalOp_mcode:(Ast_cocci.logicalOp, 'a) V.cmcode ->
         ?cv_mcode:(Ast_cocci.const_vol, 'a) V.cmcode ->
         ?sign_mcode:(Ast_cocci.sign, 'a) V.cmcode ->
         ?struct_mcode:(Ast_cocci.structUnion, 'a) V.cmcode ->
         ?storage_mcode:(Ast_cocci.storage, 'a) V.cmcode ->
         ?inc_mcode:(Ast_cocci.inc_file, 'a) V.cmcode ->
         ?dotsexpr:(Ast_cocci.expression Ast_cocci.dots, 'a) V.ccode ->
         ?dotsinit:(Ast_cocci.initialiser Ast_cocci.dots, 'a) V.ccode ->
         ?dotsparam:(Ast_cocci.parameterTypeDef Ast_cocci.dots, 'a) V.ccode ->
         ?dotstemplateparam:(Ast_cocci.templateParameterTypeDef
                             Ast_cocci.dots, 'a)
                            V.ccode ->
         ?dotsstmt:(Ast_cocci.statement Ast_cocci.dots, 'a) V.ccode ->
         ?dotsanndecl:(Ast_cocci.annotated_decl Ast_cocci.dots, 'a) V.ccode ->
         ?dotsannfield:(Ast_cocci.annotated_field Ast_cocci.dots, 'a) V.ccode ->
         ?dotsenumdecl:(Ast_cocci.enum_decl Ast_cocci.dots, 'a) V.ccode ->
         ?dotsdefpar:(Ast_cocci.define_param Ast_cocci.dots, 'a) V.ccode ->
         ?ident:(Ast_cocci.ident, 'a) V.ccode ->
         ?expr:(Ast_cocci.expression, 'a) V.ccode ->
         ?assignOp:(Ast_cocci.assignOp, 'a) V.ccode ->
         ?binaryOp:(Ast_cocci.binaryOp, 'a) V.ccode ->
         ?ty:(Ast_cocci.typeC, 'a) V.ccode ->
         ?ft:(Ast_cocci.fullType, 'a) V.ccode ->
         ?init:(Ast_cocci.initialiser, 'a) V.ccode ->
         ?param:(Ast_cocci.parameterTypeDef, 'a) V.ccode ->
         ?template_param:(Ast_cocci.templateParameterTypeDef, 'a) V.ccode ->
         ?define_param:(Ast_cocci.define_param, 'a) V.ccode ->
         ?decl:(Ast_cocci.declaration, 'a) V.ccode ->
         ?annotated_decl:(Ast_cocci.annotated_decl, 'a) V.ccode ->
         ?field:(Ast_cocci.field, 'a) V.ccode ->
         ?annotated_field:(Ast_cocci.annotated_field, 'a) V.ccode ->
         ?enumdecl:(Ast_cocci.enum_decl, 'a) V.ccode ->
         ?stmt:(Ast_cocci.statement, 'a) V.ccode ->
         ?rule:(Ast_cocci.rule_elem, 'a) V.ccode ->
         ?case:(Ast_cocci.case_line, 'a) V.ccode ->
         ?string_fragment:(Ast_cocci.string_fragment, 'a) V.ccode ->
         ?fmt:(Ast_cocci.string_format, 'a) V.ccode ->
         ?attribute:(Ast_cocci.attr, 'a) V.ccode ->
         ?attr_arg:(Ast_cocci.attr_arg, 'a) V.ccode ->
         ?pragma_info:(Ast_cocci.pragmainfo, 'a) V.ccode ->
         ?top:(Ast_cocci.top_level, 'a) V.ccode -> 'a Visitor_ast.combiner
This argument cannot be applied without label
make: *** [Makefile:431: ocaml/no_prepare_ocamlcocci.cmx] Error 2

**Comment by elfring** [comment]
>Trying to build the current master branch on rocky linux 8 (~centos 8)

:thought_balloon: How do you think about to offer any more background information like version numbers (for OCaml software components)?

**Comment by sahlberg** [comment]
I am ignorant aobut ocaml but I did install what I guess is the latest version yesterday from
https://opam.ocaml.org/doc/Install.html
I.e. using the binary installation script.

I tried to run coccinelle many times yesterday :
./autogen
./configure
make
and it all failed every time. Trying to add "echo" to the various makefiles to see if I could figure out what it was trying to do.

Today I tried to run
./autogen
./configure
make

and now it works every time. Lol.
The only thing I can think of is that I build on a nfsv3 share if that would have any effect.

It works now though but I have no idea why.
